test/esm.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const Code = require('@hapi/code');
+const Lab = require('@hapi/lab');
+
+
+const { before, describe, it } = exports.lab = Lab.script();
+const expect = Code.expect;
+
+
+describe('import()', () => {
+
+    let Statehood;
+
+    before(async () => {
+
+        Statehood = await import('../lib/index.js');
```

Comment:
I'm not too familiar with how you're supposed to test this, but shouldn't you import `'..'` instead, to make sure the entry points of the package work?

Reply:
Local imports must specify the exact file to import.

You would probably have to add yourself to dev-dependencies like:

```json
"devDependencies": {
  "@hapi/statehood": "file:."
}
```

and do a full `import('@hapi/statehood')` to use the reference from `package.json`.

Reply:
Is what's currently in the PR testing the same thing or are you suggesting we do that?

Reply:
Dunno – depends on whether you want to spend the effort to add the dev-dependencies now, or just import the file as it would resolve it (and remember this if it is ever moved).

FYI, adding it as a dev-dependency will make any other dependency use the current logic, vs. what should have been downloaded based on its `package.json` (when the version range matches). This might be a good thing, though it makes it possible to really break stuff when developing a module.

Reply:
Looks like it might complicate things for not much benefit.

Reply:
Oh, it seems that `npm` already creates a symlink without the dev-dependency line. Then you should be able to just do a full `import('@hapi/statehood')`.

Reply:
> Oh, it seems that `npm` already creates a symlink without the dev-dependency line.

I am not seeing that behavior with npm v8 or v7— perhaps it's a setting I don't have turned-on.  I see upsides to being able to reference the module the exact same way a consumer would.  I am going to land this as-is, and we can always revisit if we are able to land on an approach that feels good— there are already a good handful of modules that use this same approach to testing the ESM exports.